bytestring.cabal

```diff
@@ -1,3 +1,4 @@
+Cabal-Version:       2.2
```

Comment:
Could we possibly avoid this? It happens to me to test certain build configurations on Ubuntu 18.04, where `apt install cabal-install` gives you only `cabal-1.24`. For instance, `i386` build failed because of this, and I have not found a newer docker image for 32-bit platform.

There is a [workaround](https://github.com/haskell/text/blob/7a492ecff429748386dbde7da0db45a0bfb8dcda/.drone.yml#L20-L35), but it takes much longer to run, and CI configuration is fiddly enough already.

Reply:
I needed this to have `elif` in the Cabal file, as before `2.2`, this causes the build to fail due to not finding the files it needs. If we can re-work the Cabal file to not need an `elif`, then I'm happy to revert that change.

Reply:
This should work:

```
  if (arch(x86_64) || arch(i386))
      c-sources:   cbits/x86/is-valid-utf8.c
  else
    if arch(arm64)
        c-sources: cbits/aarch64/is-valid-utf8.c
    else
        c-sources: cbits/is-valid-utf8.c
```